sql-plugin/src/main/scala/org/apache/spark/sql/rapids/ExecutionPlanCaptureCallback.scala

```diff
@@ -130,6 +130,17 @@ object ExecutionPlanCaptureCallback {
     assertNotContain(executedPlan, gpuClass)
   }
 
+  def assertContainsAnsiCast(df: DataFrame): Unit = {
+    val executedPlan = ExecutionPlanCaptureCallback
+      .extractExecutedPlan(df.queryExecution.executedPlan)
+    assert(containsPlanMatching(executedPlan,
+      _.expressions.exists {
+        case Alias(e, _) => PlanShims.isAnsiCast(e)
+        case GpuAlias(e, _) => PlanShims.isAnsiCast(e)
```

Comment:
I pushed more changes. I now have `ExecutionPlanCaptureCallback` delegating to `PlanShims.isAnsiCastOptionallyAliased`. Let me know if that resolves the issue.

Reply:
LGTM. There is nothing special in verifying this fix. Successfully running a test like `test_hive_copy_ints_to_long` should suffice.

Reply:
Those tests were passing for me locally but I suppose that was only because I was building for one spark version at a given time and not using the aggregator jar. I will kick off another build in CI.